astropy/io/fits/fitsrec.py

```diff
@@ -841,9 +843,13 @@ def _convert_p(self, column, field, recformat):
                 dummy[idx] = raw_data[offset : offset + arr_len].view(dt)
                 if column.dim and len(vla_shape) > 1:
                     # The VLA is reshaped consistently with TDIM instructions
-                    vla_dim = vla_shape[:-1]
-                    vla_dimlast = int(len(dummy[idx]) / np.prod(vla_dim))
-                    dummy[idx] = dummy[idx].reshape(vla_dim + (vla_dimlast,))
+                    if vla_shape[0] == 1:
+                        dummy[idx] = dummy[idx].reshape(1, len(dummy[idx]))
```

Comment:
This is an ad hoc solution to cover the first case in the test.
I couldn't find anything better for now...